ddpui/api/airbyte_api.py

```diff
@@ -1,748 +1,329 @@
-from ddpui.ddpairbyte import airbyte_service
-from ddpui.ddpairbyte.schema import (
-    AirbyteConnectionCreate,
-    AirbyteConnectionCreateResponse,
-    AirbyteGetConnectionsResponse,
-    AirbyteConnectionSchemaUpdate,
-    AirbyteConnectionSchemaUpdateSchedule,
-    AirbyteDestinationCreate,
-    AirbyteDestinationUpdate,
-    AirbyteSourceCreate,
-    AirbyteSourceUpdate,
-    AirbyteWorkspace,
-    AirbyteWorkspaceCreate,
-    AirbyteSourceUpdateCheckConnection,
-    AirbyteDestinationUpdateCheckConnection,
-    AirbyteConnectionUpdate,
-)
-from ddpui.auth import has_permission
-
-from ddpui.models.org_user import OrgUser
-from ddpui.models.org import OrgType
-from ddpui.models.llm import LogsSummarizationType, LlmSession, LlmSessionStatus
-from ddpui.ddpairbyte import airbytehelpers
+from typing import List, Dict, Any, Optional, Union, Tuple, cast
+from pydantic import BaseModel, Field
+from django.http import JsonResponse
+from rest_framework.request import Request
+
+from ddpui.models.org import Org
+from ddpui.models.airbyte_sync import AirbyteSync, AirbyteSyncRun
+from ddpui.models.warehouse import WarehouseCredential
+from ddpui.models.credentials import DataSourceCredential
+from ddpui.ddpairbyte.airbyte_service import AirbyteService
 from ddpui.utils.custom_logger import CustomLogger
-from ddpui.celeryworkers.tasks import (
-    get_connection_catalog_task,
-    add_custom_connectors_to_workspace,
-    summarize_logs,
-)
-from ddpui.models.tasks import (
-    TaskProgressHashPrefix,
-    TaskProgressStatus,
-)
-from ddpui.utils.singletaskprogress import SingleTaskProgress
-
-airbyte_router = Router()
-logger = CustomLogger("airbyte")
-
 
-@airbyte_router.get("/source_definitions", auth=auth.CustomAuthMiddleware())
-@has_permission(["can_view_sources"])
-def get_airbyte_source_definitions(request):
-    """Fetch airbyte source definitions in the user organization workspace"""
-    orguser: OrgUser = request.orguser
-    if orguser.org.airbyte_workspace_id is None:
-        raise HttpError(400, "create an airbyte workspace first")
-
-    res = airbyte_service.get_source_definitions(orguser.org.airbyte_workspace_id)
-
-    # filter source definitions for demo account
-    allowed_sources = os.getenv("DEMO_AIRBYTE_SOURCE_TYPES")
-    if orguser.org.base_plan() == OrgType.DEMO and allowed_sources:
-        res["sourceDefinitions"] = [
-            source_def
-            for source_def in res["sourceDefinitions"]
-            if source_def["name"] in allowed_sources.split("|")
-        ]
-    logger.debug(res)
-    return res["sourceDefinitions"]
-
-
-@airbyte_router.get(
-    "/source_definitions/{sourcedef_id}/specifications",
-    auth=auth.CustomAuthMiddleware(),
-)
-@has_permission(["can_view_sources"])
-def get_airbyte_source_definition_specifications(request, sourcedef_id):
+logger = CustomLogger("airbyte_api")
+
+# Pydantic models for request and response payloads
+class SourceDefinitionResponse(BaseModel):
```

Comment:
@aryan0931 solid first attempt. 

Could we move the pydantic models to `ddpui/ddpairbyte/schema.py` and use `from ninja import Schema` (its basically the same as pydantic model) to keep consistency in the code base.  We have been using in our entire codebase